assets/express/src/services/user.service.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  STATUS,
+  UserService,
+} from './interfaces';
+
+import jwt from 'jsonwebtoken';
+import { UserPayload } from '../entities';
+import { hashPassword, compareHash } from './helpers';
+import { UserRepository } from '@database';
+
+const JWT_SECRET: string = process.env.JWT_SECRET_KEY || 'Jw7_S3Cr37K3y';
+
+export function userServiceFactory(userRepository: UserRepository): UserService {
+  async function loginUser(
+    payload: { email: string, password: string }
+  ) { 
+    const { email, password } = payload;
+    const errors = [];
+    let token;
+
+    try {
+      const user = await userRepository.getUser(email);
+      console.log({ user });
+      if (!user) {
+        throw new Error('Wrong email or password');
+      }
+
+      const passwordCheck = await compareHash(password, user.password);
+
+      if (!passwordCheck) {
+        throw new Error('Wrong email or password');
+      }
+
+      token =
+        jwt.sign(
+          {
+            email: email,
+            role: user.role,
+          },
+          JWT_SECRET
+        ) || null;
+    } catch (err) {
+      if (err instanceof Error) {
+        errors.push(err.message);
+      } else {
+        errors.push('Unknown Error');
+      }
+    }
+
+    const data = !errors.length && token ? [{ token }] : [];
+
+    return {
+      errors,
+      status: data.length ? STATUS.SUCCESS : STATUS.FAILURE,
+      data,
+    };
+  }
+
+  async function createUser(
+    payload: UserPayload
+  ) {
+    const { email, password, ...attributes } = payload;
+    console.log({ payload });
+    const errors: Error[] = [];
+    let user;
+
+    try {
+      const hashedPassword = await hashPassword(10, password);
+      user = await userRepository.createUser({
+        email,
+        password: hashedPassword,
+        ...attributes,
+        role: 'user',
+      });
+    } catch (err) {
+      errors.push(err as Error);
+    }
+
+    return {
+      errors,
+      status: !errors.length && user ? STATUS.SUCCESS: STATUS.FAILURE,
```

Comment:
The status is something related to the route itself, not to the service. Services shouldn't know about routes in general, they just do the business logic.